### ISSUE TITLE
Leia and Artwork

### ISSUE THREAD

**Issue by aerogems** [croneter/PlexKodiConnect#509]
I have been having a persistent issue with PKC downloading artwork for TV shows. The main cover (poster) art, not fanart or clearart.

I'm using Leia Alpha 2 (not a nightly) the latest PKC beta (though I've had this issue with the stable version as well) and PMS 1.13.4.5251, but this same issue has happened with previous versions as well. PMS is running on a Synology NAS, so effectively Linux, and Kodi is running on a Shield TV (so Android TV Oreo). This is all over a gigabit LAN connection.

Last night I discovered the latest Synology update is buggy and prevents PMS from running, forcing me to reinstall the OS. It also ended up forcing me to rebuild the Kodi database using PKC. Since then, I have been unable to get any artwork to appear for any of my TV shows. I left the Shield TV running Kodi for at least the last five hours, and still not a single show has artwork in Kodi. If I go to the PMS admin page, every show has artwork.

This has happened before, and if I revert back to Kyrpton, it will download all of the artwork in parallel with the database sync. Leia just shows a black box with a "play" triangle in a circle. I can then upgrade to Leia and the artwork will carry over, but it will not always show artwork for any new shows I may add.

If I go into the PKC artwork settings and tell it to cache all the images to the Kodi texture cache, I get an error saying the thread crashed.

Let me know if there's any additional details you want/need. I will do my best to provide them.

**Comment by croneter** [comment]
Ough, sorry to tell you that this stinks of a Leia Alpha 2 issue. "Alpha" as in "it's still under development and a lot can go wrong", never mind whether you're using PKC or not. There's probably not a lot I can do at the moment. 

Could you please
* enable Kodi debug logging in the Kodi system settings
* Then reset the Kodi database via PKC settings - Advanced - Reset (in red). Hit YES, then NO. 
* Let Kodi reboot and PKC resync. Let Kodi sit for at least 30min. 
* Get the Kodi debug log file and share it here, as described here: https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug

**Comment by aerogems** [comment]
The log file is attached. Since it was rather large, I put it into a ZIP archive.

One minor detail I forgot to mention earlier, is that with Krypton, I do not need to wait for the artwork to download. It is all there when I finish the library sync. Just the same, I did let Kodi sit for a while as you requested.

And I know Alpha means it is a very early release. I actually upgraded to see if a particular Kodi issue was with Kodi or Android TV (sadly it was Android TV, where skipping in a file with passthrough audio like dolby surround, you lose audio until you either restart Kodi or turn passthrough off and on). If you can't fix this with Leia, I can go back to Krypton no problem, but if you can save me the hour or so it takes to copy my old Kodi 17 profile back to my Shield, that's even better.

[kodi.zip](https://github.com/croneter/PlexKodiConnect/files/2206737/kodi.zip)

**Comment by croneter** [comment]
I see that Kodi complains seeing invalid SSL certificates:
```
ERROR: CCurlFile::Stat - Failed: SSL peer certificate or SSH remote key was not OK(51) for https://192.168.1.21:32400/photo/:/transcode?width=3840&height=3840&minSize=1&upscale=0&url=/library/metadata/12738/thumb/1531908039&X-Plex-Token=XXX```

Try the following:
* Go to the PlexKodiConnect settings, `Connection`. Choose the first option to list your Plex Media Servers. Pick one
* Let PKC reconnect. 
* Restart Kodi to be on the safe side
* Go to the PKC settings, `Advanced`, then reset the Kodi database (in red). Hit YES, then NO. 
* Let PKC resync

**Comment by croneter** [comment]
You did not follow https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug and shared your secret Plex token. Please **IMMEDIATELY** follow [I published my Plex token to some forum or Github. Anyone can now access my Plex Server!](https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug#i-published-my-plex-token-to-some-forum-or-github-anyone-can-now-access-my-plex-server)

**Comment by croneter** [comment]
I guess this issue is solved. Please reopen / post here if this is not the case

**Comment by aerogems** [comment]
I went back to 17.6 where everything works great with the exact same PKC settings, but now that Alpha 3 is out, I decided to give Leia another try, and am having exactly the same issue.

- I made sure PKC is searching for the server and using that instead of a hardcoded IP
- When I connect to my PMS via the web, I get the green padlock icon for a secure connection
- I tried turning off the SSL option in PKC

One thing I will say, is that if I try to connect to my PMS via a direct IP address, it complains about the SSL cert not being valid, but if I go through the app.plex.tv URL, it works fine. I'm running PMS on a Synology NAS, and I've tried setting up a self-signed SSL cert, but for the life of me I cannot get any browsers to accept it, even after importing the key into the trusted CA. 

So, why would everything work perfectly with 17.6, but then the exact same settings fail on18.0a2 and a3? Can PKC use the app.plex.tv URL?

**Comment by coz2001** [comment]
I had the same issue you are having with missing posters but only on content added after kodi & pkc setup.
I tried with both alpha2 and some nightlyes weeks in between, then when alpha3 was released I was curious and tried again this time I haven't had a single problem, I know this isen't any help with your issue just wan't share my experience.

Hope it will get solved some how.

**Comment by croneter** [comment]
@aerogems 
Well, for one thing Kodi 18 is still an Alpha version and thus still very unstable. Please share a DEBUG log file of you resetting the Kodi database (under Advanced, last setting in red; hit YES, then NO) and PKC then rescanning everything. See here: https://github.com/croneter/PlexKodiConnect/wiki/How-to-Report-A-Bug

**Comment by aerogems** [comment]
@croneter 
I am aware that Leia is still considered alpha level software and what that means. I'm not blaming you for anything, just hoping I can help isolate an issue so either you can figure out how to fix it in your code, or it can be reported to the Kodi devs as an issue with the libcurl module. Did Kodi maybe update the version of libcurl they are using and it changed some of the function arguments?

Moving on... Ran into a problem trying to get you a debug log. When I enable verbose logging it causes the PKC settings dialog to fail to display properly. I am going to try with a brand new profile to see if that magically makes a difference. I will edit this post, or make a new one, when that is done.

However, looking at the log that was generated, I see a large number of lines like this

**11:17:40.789 T:139478500592   DEBUG: CAddonSettings[plugin.video.plexkodiconnect]: trying to load setting definitions from old format...
11:17:40.792 T:139478500592   DEBUG: CAddonSettings[plugin.video.plexkodiconnect]: loading setting values
11:17:40.793 T:139478500592   DEBUG: CSettingsManager: requested setting (caching_artwork_count) was not found.**

and

**11:18:01.608 T:139415504112   DEBUG: PLEX.librarysync: Analyzed all seasons of TV show with Plex Id 14262
11:18:01.626 T:139268654320   DEBUG: DatabaseUtils::GetSortFieldList: unknown field 31**

Those are the only things that look like an error that showed up in the log since I could not reset the Kodi/PKC database and rerun the scan.

**Comment by aerogems** [comment]
[kodi.zip](https://github.com/croneter/PlexKodiConnect/files/2271844/kodi.zip)

@croneter 
I have attached the Kodi debug log. I also enabled the component logging for libcurl since that is where the SSL errors keep coming from. This log starts after Kodi restarts to re-sync the library with Plex. I waited until I saw the notification that PKC was done running the FanartTV scrape before exiting Kodi to get the log. 

The results were the same as before. It downloads everything except the artwork without issue.

**Comment by croneter** [comment]
@aerogems 
PKC still uses a local IP `192.168.1.21` instead of `plex.direct`. 
```
CCurlFile::Stat - Failed: SSL peer certificate or SSH remote key was not OK(51) for https://192.168.1.21:32400/photo/:/transcode?width=3840&height=3840&minSize=1&upscale=0&url=/library/metadata/14564/thumb/1530856414&X-Plex-Token=DELETED
```
This is a KODI thing, nothing PKC can control. KODI checks the certificate for your PMS, which works only for `plex.direct`, and sees that it does not work for `192.168.1.21`. 

Could you please make sure that debug logging is enabled, restart Kodi, then go to the PKC settings, `Connection`, and `Choose PMS from a list` (first setting). Let it do it's thing. Reconnect to your PMS. Then share your log again. I should be able to see what's going on and why PKC does not use `plex.direct`

**Comment by aerogems** [comment]
@croneter 

I kept a copy of the kodi.old.log file which should have info from when I first set up PKC. If you still need more, I'll reinstall 18.0a3 in the morning and get you some new logs.

[kodi.old.zip](https://github.com/croneter/PlexKodiConnect/files/2272970/kodi.old.zip)

**Comment by croneter** [comment]
@aerogems 
Looking at the log, you (nor PKC) did not re-connect to the PMS. Hence the local connection stayed `192.168.1.21`. If you run into this issue again, simply reconnect to your PMS by going into the PKC settings, `Connection`, then select `Choose a PMS from a list` (or some such)

**Comment by croneter** [comment]
Well there is nothing in the log indicating that you chose the PMS again. Since I understood you're not using Kodi 18 at the moment, don't worry about it. If you switch back again to Kodi 18 and it's still an issue, please share a Kodi 18 log including the choosing-of-the-PMS 😉 

If you still got Kodi 18 up and running though, I need the debug log of you choosing the PMS again.

**Comment by croneter** [comment]
@aerogems 
Plus: can you access http://plex.direct/ from your Nvidia Shield and/or plex.tv? (even if it's just telling you "404 Not Found")

**Comment by aerogems** [comment]
@croneter 
http://plex.direct works (404 ngix page) and I'm trying to sideload a browser that supports https. Will edit this post with more details when done.

I have also reinstalled Kodi 18.0a3 and will do a complete setup of PKC with debugging on, assuming the PKC settings page doesn't fail to render like last time. I will provide those details when I have them.

**Comment by aerogems** [comment]
Here is a log starting with a profile migrated from 17.6. I then went to the advanced section and did the complete reset of all PKC settings and the Kodi database. Kodi restarted, I went through the whole initial setup thing, picked the local PMS.... no change at the end of it all. I took some photos of my steps along the way if you want me to post them as well.

I included both kodi.log and kodi.old.log this time since I'm not sure how exactly it is Kodi decides when it's time to switch from one to the other.

[kodi-8-9-18-1pm.zip](https://github.com/croneter/PlexKodiConnect/files/2275755/kodi-8-9-18-1pm.zip)

**Comment by aerogems** [comment]
@croneter 
I solved it. Good news is, PKC wasn't at fault. After some more googling, I came across [this post on Reddit](https://www.reddit.com/r/OnHub/comments/3za5gv/dns_rebinding_issue_connecting_to_local_secure/cz4vhd1).

For whatever stupid reason, I had to manually set my Shield TV's DNS settings to match those of my router instead of relying on DHCP. Once I did that, I got a plex.direct address when I had PKC check for servers. I reset the Kodi database after installing 18.0a3, and am now getting all artwork. Hopefully this information will help someone in the future. Appreciate all your efforts chasing this down.
